platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetsPlugin.java

```diff
@@ -67,24 +72,41 @@
     private static final int FILE_CREATION_ERROR = 7;
     private static final int JSON_CREATION_ERROR = 8;
 
-    private String pathToStorage;
+    private String pathToDatabase;
+    private String pathToAssets;
     private int blockSize;
 
     @Override
     public void initialize(CordovaInterface cordova, CordovaWebView webView) {
         super.initialize(cordova, webView);
 
         final Context applicationContext = cordova.getActivity().getApplicationContext();
-        pathToStorage = applicationContext.getCacheDir().getAbsolutePath() + File.separator;
+
+        pathToDatabase = applicationContext.getCacheDir().getAbsolutePath() + File.separator + PLUGIN_FOLDER;
+        pathToAssets = pathToDatabase + File.separator + ASSETS_FOLDER;
+
+        if (!createFolderIfRequired(pathToDatabase)) {
+            Log.e(TAG, "error -- unable to create folder: " + pathToDatabase);
+            return;
+        } else if (!createFolderIfRequired(pathToAssets)) {
+            Log.e(TAG, "error -- unable to create folder: " + pathToAssets);
+            return;
+        }
+
+        pathToDatabase += File.separator;
+        pathToAssets += File.separator;
+
         setBlockSize();
 
-        wizAssetManager = new WizAssetManager(applicationContext);
+        wizAssetManager = new WizAssetManager(applicationContext, pathToDatabase);
+
+        initialized = wizAssetManager.isReady();
```

Comment:
I think `isReady` is not useful here. If the database could not have been opened, an exception would have been thrown normally no?